symengine/functions.cpp

```diff
@@ -2703,6 +2718,142 @@ RCP<const Basic> uppergamma(const RCP<const Basic> &s, const RCP<const Basic> &x
+        (rcp_static_cast<const Rational>(y)->i.get_den()) == 2)) {
+            return false;
+        }
+    }
+    return true;
+}
+
+std::size_t Beta::__hash__() const
+{
+    std::size_t seedx = BETA;
+    std::size_t seedy = BETA;
+    hash_combine<Basic>(seedx, *x_);
+    hash_combine<Basic>(seedy, *y_);
+    return seedx^seedy;
+}
+
+bool Beta::__eq__(const Basic &o) const
+{
+    if (is_a<Beta>(o)) {
+        if(eq(*x_, *(static_cast<const Beta &>(o).x_)) and
+        eq(*y_, *(static_cast<const Beta &>(o).y_)))
+            return true;
+        if(eq(*y_, *(static_cast<const Beta &>(o).x_)) and
+        eq(*x_, *(static_cast<const Beta &>(o).y_)))
+            return true;
+    }
+    return false;
+}
+
+int Beta::compare(const Basic &o) const
+{
+    SYMENGINE_ASSERT(is_a<Beta>(o))
+    const Beta &b = static_cast<const Beta &>(o);
+    if (eq(*x_, *(b.x_))) {
+        return y_->__cmp__(*(static_cast<const Beta &>(o).y_));
+    }
+    return x_->__cmp__(*(static_cast<const Beta &>(o).x_));
+}
+
+RCP<const Basic> Beta::rewrite_as_gamma() const
+{
+    return div(mul(gamma(x_), gamma(y_)), add(x_, y_));
+}
+
+RCP<const Basic> beta(const RCP<const Basic> &x, const RCP<const Basic> &y)
+{
+    // Only special values are being evaluated
+    if(eq(*add(x, y), *one)) {
+        throw std::runtime_error("Complex Infinity not yet implemented");
+    }
+
+    if (is_a<Integer>(*x)) {
+        RCP<const Integer> x_int = rcp_static_cast<const Integer>(x);
+        if (x_int->is_positive()) {
+            if (is_a<Integer>(*y)) {
+                RCP<const Integer> y_int = rcp_static_cast<const Integer>(y);
+                if (y_int->is_positive()) {
+                    return div(mul(gamma_positive_int(x), gamma_positive_int(y)), gamma_positive_int(add(x, y)));
+                }
+                else{
+                    throw std::runtime_error("Complex Infinity not yet implemented");
+                }
+            }
+            else if(is_a<Rational>(*y)) {
+                RCP<const Rational> y_ = rcp_static_cast<const Rational>(y);
+                if ((y_->i.get_den()) == 2) {
+                    return div(mul(gamma_positive_int(x), gamma_multiple_2(y)), gamma_multiple_2(add(x, y)));
+                }
+                else {
+                    return Beta::from_two_basic(x, y);
+                }
+            }
+        }
+        else{
+            throw std::runtime_error("Complex Infinity not yet implemented");
+        }
+    }
+
+    if (is_a<Integer>(*y)) {
+        RCP<const Integer> y_int = rcp_static_cast<const Integer>(y);
+        if (y_int->is_positive()) {
+         if(is_a<Rational>(*x)) {
+                RCP<const Rational> x_ = rcp_static_cast<const Rational>(x);
+                if ((x_->i.get_den()) == 2) {
+                    return div(mul(gamma_positive_int(y), gamma_multiple_2(x)), gamma_multiple_2(add(x, y)));
+                }
+                else {
```

Comment:
These should be formatted according to https://github.com/symengine/symengine/blob/master/doc/style_guide.md#whitespace